### PR TITLE
Add Docker support

### PR DESCRIPTION
**Pull request by jess-sys**
Docker and docker-compose support

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FROM node:12
```
Comment (super-bunny) [inline Dockerfile]: Node 14 is the new LTS version, please use it instead.

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+version: "3.7"
+
+services:
+    api:
+        build:
+            context: .
+        ports:
+            - '3000:${PORT}'
```
Comment (super-bunny) [inline docker-compose.yml]:
```suggestion
            - "3000:${PORT:-3000}"
```
Put a default value for environment variable.

**Review by super-bunny — CHANGES_REQUESTED** [review]
:warning: You forgot to copy the source files in Dockerfile :scream:

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+
+WORKDIR /usr/src/app
+COPY package*.json ./
+
+RUN npm install
+
```
Comment (super-bunny) [inline Dockerfile:10]:
```suggestion

EXPOSE 3000

```
You can expose the app port for greater ease of use.

File: Dockerfile (hunk 1 above)
Comment (jess-sys) [reply Dockerfile]: Added

**Comment by jess-sys** [comment]
Requested changes implemented.

**Review by super-bunny — APPROVED** [review]
Good :+1:

**Pull request merged**